### PR TITLE
Modified VideoViewController to display the correct video

### PR DESCRIPTION
**Pull request by bjrne**

### CODE REVIEW

File: xikolo-ios/VideoViewController.swift [hunk 1]
```diff
@@ -15,24 +15,55 @@ class VideoViewController: UIViewController {
...
+    @IBOutlet weak var nextItemButton: UIButton!
+    @IBOutlet weak var containerHeighConstraint: NSLayoutConstraint!
     
-    var courseItem: CourseItem! {
+    var courseItem: CourseItem!
+    var video: Video? = nil {
         didSet {
-            updateUI()
+            videoDidLoad()
+            performSegueWithIdentifier("EmbedAVPlayer", sender: video)
```
Comment (invliD) [inline xikolo-ios/VideoViewController.swift]: Please don't trigger a segue in a didSet trigger. That's really confusing.

File: xikolo-ios/VideoViewController.swift [hunk 2]
```diff
@@ -15,24 +15,55 @@ class VideoViewController: UIViewController {
...
+        super.viewDidLoad()
+        containerHeighConstraint.constant = self.view.frame.size.width * (9/16)
+        let videoIncomplete = courseItem.content as! Video
+        VideoHelper.syncVideo(videoIncomplete).onSuccess { videoComplete in
+            self.video = videoComplete
+        }
+    }
+
+    func videoDidLoad() {
+        titleLabel.text = courseItem?.title ?? "default title"
```
Comment (invliD) [inline xikolo-ios/VideoViewController.swift]: Remove the default title.

File: xikolo-ios/VideoViewController.swift [hunk 3]
```diff
@@ -38,7 +38,7 @@ class VideoViewController: UIViewController {
     }
 
     func videoDidLoad() {
-        titleLabel.text = courseItem?.title ?? "default title"
+        titleLabel.text = courseItem?.title ?? ""
     }
```
Comment (invliD) [inline xikolo-ios/VideoViewController.swift]: Please **remove** the default, aka only leave `courseItem?.title`.

File: xikolo-ios/VideoViewController.swift [hunk 4]
```diff
@@ -15,24 +15,55 @@ class VideoViewController: UIViewController {
...
+        super.viewDidLoad()
+        containerHeighConstraint.constant = self.view.frame.size.width * (9/16)
+        let videoIncomplete = courseItem.content as! Video
+        VideoHelper.syncVideo(videoIncomplete).onSuccess { videoComplete in
+            self.video = videoComplete
+        }
+    }
+
+    func videoDidLoad() {
+        titleLabel.text = courseItem?.title
```
Comment (invliD) [inline xikolo-ios/VideoViewController.swift]: You have defined `courseItem` as `CourseItem!`, an implicitly unwrapped optional. If there's a need to check for `nil` here (using `?`), it should be a real optional. If there isn't, remove the `?`.

**Comment by invliD** [comment]
Great! Now please squash and rebase.

**Comment by bjrne** [comment]
i didn't rebase yet, because my master build is failing (TravisCI as well)

File: xikolo-ios/VideoViewController.swift [hunk 5]
```diff
@@ -15,24 +15,55 @@ class VideoViewController: UIViewController {
...
+    @IBOutlet weak var previousItemButton: UIButton!
+    @IBOutlet weak var nextItemButton: UIButton!
+    @IBOutlet weak var containerHeighConstraint: NSLayoutConstraint!
     
-    var courseItem: CourseItem! {
+    var courseItem: CourseItem!
+    var video: Video? = nil {
         didSet {
-            updateUI()
+            videoDidLoad()
```
Comment (invliD) [inline xikolo-ios/VideoViewController.swift]: Don't do this! I'm not sure if it's currently a problem, but do not call a VC's life cycle methods manually.

Comment (bjrne) [inline xikolo-ios/VideoViewController.swift]: I don't. It's my own method. Should I rename it instead?

File: xikolo-ios/VideoViewController.swift [hunk 6]
```diff
@@ -15,24 +15,55 @@ class VideoViewController: UIViewController {
...
     @IBOutlet weak var titleLabel: UILabel!
     @IBOutlet weak var descriptionTextView: UITextView!
+    @IBOutlet weak var openSlidesButton: UIButton!
+    @IBOutlet weak var previousItemButton: UIButton!
+    @IBOutlet weak var nextItemButton: UIButton!
+    @IBOutlet weak var containerHeighConstraint: NSLayoutConstraint!
     
-    var courseItem: CourseItem! {
+    var courseItem: CourseItem!
+    var video: Video? = nil {
```
Comment (invliD) [inline xikolo-ios/VideoViewController.swift]: I think `= nil` is unnecessary here, the value of optionals defaults to `nil`.

File: xikolo-ios/VideoViewController.swift [hunk 7]
```diff
@@ -15,24 +15,55 @@ class VideoViewController: UIViewController {
...
+        containerHeighConstraint.constant = self.view.frame.size.width * (9/16)
+        let videoIncomplete = courseItem.content as! Video
+        VideoHelper.syncVideo(videoIncomplete).onSuccess { videoComplete in
+            self.video = videoComplete
+            self.performSegueWithIdentifier("EmbedAVPlayer", sender: self.video)
+        }
+    }
+
+    func videoDidLoad() {
+        titleLabel.text = courseItem.title
```
Comment (invliD) [inline xikolo-ios/VideoViewController.swift]: Why don't you do this in `viewDidLoad` instead? The `courseItem` is there before the view controller is even loaded, right?

Comment (bjrne) [inline xikolo-ios/VideoViewController.swift]: Thats right, I didn't think about this one again after removing the rest earlier.

**Comment by invliD** [comment]
👍, but you need to rebase.

**Pull request merged**